src/widgets/modals/StocktakeBatchModal.js

```diff
@@ -226,21 +239,38 @@ export default class StocktakeBatchModal extends React.Component {
     );
   };
 
-  renderReasonModal = () => {
-    const { reasonModalOpen, currentBatch, reasons } = this.state;
-    const { option } = currentBatch;
-    const currentReasonIndex = option && reasons.findIndex(reason => reason.id === option.id);
+  getModalTitle = () => {
+    const { modalKey } = this.state;
+    const { REASON_EDIT } = MODAL_KEYS;
+    switch (modalKey) {
+      default:
+        return '';
+      case REASON_EDIT:
+        return programStrings.select_a_reason;
+    }
+  };
 
-    return (
-      <GenericChooseModal
-        isOpen={reasonModalOpen}
-        data={reasons}
-        highlightIndex={currentReasonIndex}
-        onPress={this.reasonModalConfirm}
-        keyToDisplay="title"
-        title={modalStrings.select_a_reason}
-      />
-    );
+  renderModalContent = () => {
+    const { modalKey, currentBatch, reasons } = this.state;
+    const { option } = currentBatch;
+    const highlightValue = option && option.title;
+    const { REASON_EDIT } = MODAL_KEYS;
+    switch (modalKey) {
+      default: {
```

Comment:
Stylistic choice, but I'd have put the default/null return as the last clause in the switch.

Reply:
c8198114c3267b88faad8799ce442c63694058cf